dotnet/src/Agents/Abstractions/AgentChat.cs

```diff
@@ -223,13 +252,19 @@ async Task<AgentChannel> GetOrCreateChannelAsync()
             AgentChannel channel = await this.SynchronizeChannelAsync(channelKey, cancellationToken).ConfigureAwait(false);
             if (channel == null)
             {
-                channel = await agent.CreateChannelAsync(cancellationToken).ConfigureAwait(false);
+                this.Logger.LogDebug("[{MethodName}] Creating channel for {AgentType}: {AgentId}", nameof(InvokeAgentAsync), agent.GetType(), agent.Id);
+
+                channel = await agent.CreateChannelAsync(this.LoggerFactory.CreateLogger(agent.GetType()), cancellationToken).ConfigureAwait(false);
+                channel.Logger = this.LoggerFactory.CreateLogger(channel.GetType());
```

Comment:
Since you are passing `this.LoggerFactory.CreateLogger(agent.GetType())` to `CreateChannelAsync` method, why it's needed to set `channel.Logger` in the next line?

Reply:
One is "agent" typed.  The other is "channel" typed.  It translates into the capture diagnostics.

Reply:
Will add comments.

Reply:
Updated